dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,331 @@
+        kpiKey: kpiKey,
+        kpiLabel: kpiField?.label ? kpiField.label : kpiKey,
+        kpiDescription: kpiField?.description ? kpiField.description : "",
+        kpiFormFieldComponent: kpiField?.component ?? "",
+        content: { [dataIdOfP2pDataset]: (kpiField, kpiValue) },
+      } as KpiDataObject;
+      if (this.mapOfKpiKeysToDataObjects.has(kpiKey)) {
+        Object.assign(kpiData.content, this.mapOfKpiKeysToDataObjects.get(kpiKey)?.content);
+      }
+      this.mapOfKpiKeysToDataObjects.set(kpiKey, kpiData);
+      this.resultKpiData = kpiData;
+    },
+    /**
+     * Retrieves and converts the stored array of P2P datasets in order to make it displayable in the frontend.
+     */
+    convertP2pDataToFrontendFormat(): void {
+      if (this.p2pDataAndMetaInfo.length) {
+        this.p2pDataAndMetaInfo.forEach((oneP2pDataset: DataAndMetaInformationPathwaysToParisData) => {
+          const dataIdOfP2pDataset = oneP2pDataset.metaInfo?.dataId ?? "";
+          const reportingPeriodOfP2pDataset = oneP2pDataset.metaInfo?.reportingPeriod ?? "";
+          this.arrayOfReportingPeriodWithDataId.push({
+            dataId: dataIdOfP2pDataset,
+            reportingPeriod: reportingPeriodOfP2pDataset,
+          });
+          for (const [categoryKey, categoryObject] of Object.entries(oneP2pDataset.data) as [string, object] | null) {
+            if (categoryObject == null) continue;
+            const listOfDataObjects: Array<KpiDataObject> = [];
+            const frameworkCategoryData = assertDefined(p2pDataModel.find((category) => category.name === categoryKey));
+            this.iterateThroughSubcategories(
+              categoryObject,
+              categoryKey,
+              frameworkCategoryData,
+              dataIdOfP2pDataset,
+              listOfDataObjects,
+              oneP2pDataset.data
+            );
+
+            this.mapOfCategoryKeysToDataObjectArrays.set(frameworkCategoryData.label, listOfDataObjects);
+          }
+        });
+      }
+      this.arrayOfReportingPeriodWithDataId = sortReportingPeriodsToDisplayAsColumns(
+        this.arrayOfReportingPeriodWithDataId as ReportingPeriodOfDataSetWithId[]
+      );
+    },
+    /**
+     * Iterates through all subcategories of a category
+     * @param categoryObject the data object of the framework's category
+     * @param categoryKey the key of the corresponding framework's category
+     * @param frameworkCategoryData  the category object of the framework's category
+     * @param dataIdOfP2pDataset  the data ID of the P2P dataset
+     * @param listOfDataObjects a map containing the category and it's corresponding Kpis
+     * @param oneP2pDataset dataset for which the show if conditions should be checked
+     */
+    iterateThroughSubcategories(
+      categoryObject,
+      categoryKey,
+      frameworkCategoryData: Category,
+      dataIdOfP2pDataset: string,
+      listOfDataObjects: Array<KpiDataObject>,
+      oneP2pDataset: PathwaysToParisData
```

Comment:
Why "one"P2pDataset?

Reply:
Because it is the one dataset determined by the forEach loop

Reply:
The parameters of iterateA and iterateB still have this name.

Reply:
Now they should have the new name...